### PR TITLE
Helm chart for harbor-config-operator

### PR DESCRIPTION
**Pull request by DeeAjayi**

### CODE REVIEW

File: helm/harbor-config/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+      containers:
+      - args:
+        - --secure-listen-address=0.0.0.0:8443
+        - --upstream=http://127.0.0.1:8080/
+        - --logtostderr=true
+        - --v=0
+        env:
+        - name: KUBERNETES_CLUSTER_DOMAIN
+          value: {{ .Values.kubernetesClusterDomain }}
+        image: {{ .Values.controllerManager.kubeRbacProxy.image.repository }}:{{ .Values.controllerManager.kubeRbacProxy.image.tag
```
Comment (MarcelMue) [inline helm/harbor-config/templates/deployment.yaml:38]: IS this rbac proxy really needed? I personally don't think it should be.

File: helm/harbor-config/templates/harborconfiguration-crd.yaml [hunk 2]
```diff
@@ -0,0 +1,127 @@
+apiVersion: apiextensions.k8s.io/v1
```
Comment (MarcelMue) [inline helm/harbor-config/templates/harborconfiguration-crd.yaml:1]: We will need to see how we handle this CRD. Having it in templates will likely cause trouble.

File: helm/harbor-config/templates/manager-config.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  labels:
+  {{- include "harbor-config.labels" . | nindent 4 }}
+data:
+  controller_manager_config.yaml: |
+    apiVersion: controller-runtime.sigs.k8s.io/v1alpha1
+    health:
+      healthProbeBindAddress: {{ .Values.managerConfig.controllerManagerConfigYaml.health.healthProbeBindAddress
+        | quote }}
+    kind: ControllerManagerConfig
+    leaderElection:
```
Comment (MarcelMue) [inline helm/harbor-config/templates/manager-config.yaml:14]: We will likely only run a single replica - not sure if exposing leader election makes sense.

File: helm/harbor-config/values.yaml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        memory: 64Mi
+  replicas: 1
+kubernetesClusterDomain: cluster.local
+managerConfig:
+  controllerManagerConfigYaml:
+    health:
+      healthProbeBindAddress: :8081
+    leaderElection:
+      leaderElect: true
+      resourceName: c8ffa958.harbor.configuration
```
Comment (MarcelMue) [inline helm/harbor-config/values.yaml:32]: Seems very specific, like from a test?

File: helm/harbor-config/templates/deployment.yaml (hunk 1 above)
Comment (DeeAjayi) [reply helm/harbor-config/templates/deployment.yaml]: This is what the manifest file says it does `This patch inject a sidecar container which is a HTTP proxy for the controller manager, it performs RBAC authorization against the Kubernetes API using SubjectAccessReviews.`

I have removed it though

**Comment by DeeAjayi** [comment]
@MarcelMue I have removed the leadereletion and kubeproxy settings. I also included the VPA config

File: config/manager/manager.yaml [hunk 5]
```diff
@@ -50,11 +48,20 @@ spec:
...
-          limits:
-            cpu: 500m
-            memory: 128Mi
           requests:
             cpu: 10m
             memory: 64Mi
       serviceAccountName: controller-manager
       terminationGracePeriodSeconds: 10
+---
+apiVersion: autoscaling.k8s.io/v1
```
Comment (MarcelMue) [inline config/manager/manager.yaml:57]: Separate file please

File: helm/harbor-config/templates/controller-manager-vpa.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+  name: {{ include "harbor-config.fullname" . }}-controller-manager-vpa
+  labels:
+  {{- include "harbor-config.labels" . | nindent 4 }}
+spec:
+  targetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: {{ include "harbor-config.fullname" . }}-controller-manager
+  updatePolicy:
+    updateMode: Auto
```
Comment (MarcelMue) [inline helm/harbor-config/templates/controller-manager-vpa.yaml:13]: Please make sure to have newlines at the end of all files.

File: helm/harbor-config/templates/harborconfiguration-crd.yaml (hunk 2 above)
Comment (DeeAjayi) [reply helm/harbor-config/templates/harborconfiguration-crd.yaml:1]: What'll be the recommended approach to handling CRDs?

Comment (MarcelMue) [reply helm/harbor-config/templates/harborconfiguration-crd.yaml:1]: I would personally prefer this way:
1. Use the builtin CRD folder for helm https://github.com/giantswarm/kyverno-app/tree/main/helm/kyverno/crd
2. Use a job to keep them up to date (helm only does install time apply) https://github.com/giantswarm/kyverno-app/tree/main/helm/kyverno/templates/crd-install

Examples taken from the kyverno app

Comment (MarcelMue) [reply helm/harbor-config/templates/harborconfiguration-crd.yaml:1]: I am fine to do this in a follow up - so we can get this merged quicker. Just create an issue please.

Comment (DeeAjayi) [reply helm/harbor-config/templates/harborconfiguration-crd.yaml:1]: Alright, issue created - [here](https://github.com/giantswarm/harbor-config-operator/issues/16)

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**